### DISCUSSION TITLE
Performing http requests in the frontend

### DISCUSSION THREAD

**Discussion by bobwatcherx** [zauberzeug/nicegui#638] [Q&A] [answered]
I have something very specific to ask, since nicegui is made upon fastapi, do you think will pe possible to use the button on frontend to retrieve a get method of fastapi? for example i have a simple endpoint which make a sum of two floats, would be possible with nicegui to send these inputs to the endpoint and then to display sum result on the nicegui's frontend?

**Comment by rodja** [comment, marked as answer]
Of course you can perform requests from the frontend. But you need to do them in JavaScript. So it's a little bit cumbersome.

NiceGUI follows a "backend-first" strategy. That means it will generate the frontend for you and handle all the communication. So in normal circumstances it would be much better to extract the code behind the FastAPI endpoint into function which you can then simply call when the user clicks a button or similar. Or, if you do not have the code under control (for example because it's in an external library or different server) you can do the http request in the backend.
